proposed/http-factory/http-factory.md

```diff
@@ -163,14 +170,17 @@ fwrite($resource, $body);
 Has the ability to create streams for uploaded files.
 
 ```php
-namespace Psr\Http\Message;
+namespace Psr\Http\Factory;
+
+use Psr\Http\Message\UploadedFileInterface;
 
 interface UploadedFileFactoryInterface
 {
     /**
      * Create a new uploaded file.
      *
-     * If a string is passed it is assumed to be a file path.
+     * If a string is used to create the file, a temporary resource will be
+     * created with the content of the string.
```

Comment:
I think the `$error` parameter is likely not useful here when creating these instances manually; if you're creating it manually, you were likely able to create the stream, which means the upload was successful. The only reason it might _not_ be is if there was a mismatch between things like content-length and actual stream length.

In terms of a system that does not have `$_FILES`, you may want to be able to drop in middleware for handling file uploads that was written agnostic of the environment it'd be in. This would allow it to be used in either a standard SAPI, or with an async server (which parses the incoming request manually in order to create the request instance). In this latter case, it may be useful to delay parsing of uploaded files unless absolutely needed, in order to reduce both CPU and memory overhead; in such a case, _middleware_ might do the actual parsing.

BTW, `$_FILES` is not relevant even in standard SAPIs in one key circumstance: non-POST requests will not populate it! As such, having middleware that can parse the parts and create uploaded file instances is key.